### PR TITLE
fix(deck-layers): align source integration with latest APIs

### PR DESCRIPTION
**Pull request by ibgreen**
## Goals

- Keep unified SourceLayer metadata discovery aligned with the latest loaders.gl source contracts.
- Make raster rendering honor current CRS and no-data metadata without adding application glue.
- Preserve deterministic, upstreamable behavior with focused regression coverage.

## Changes

- Normalize authority-coded PROJJSON CRS definitions into shared source view hints and raster viewport/rendering paths.
- Make nested metadata bounds follow the same depth-first semantics as automatic first-leaf layer selection.
- Ensure named group layers are not selected when only a descendant leaf can be rendered.
- Apply metadata-level no-data values when raster responses omit the repeated value.
- Improve unsupported-CRS diagnostics and document the latest raster metadata behavior.
- Add browser tests covering nested metadata/CRS discovery, explicit empty layer selections, CRS objects, and metadata no-data transparency.

## Validation

- `yarn lint fix`
- `yarn test-audit` — 609 files, 0 Tape violations
- `node_modules/.bin/tspc --build modules/deck-layers/tsconfig.json --pretty false`
- `yarn test-node` — 361 passed, 6 skipped
- `yarn build-workers`
- `yarn test-headless` — 3,105 passed, 40 skipped

`yarn build` was attempted but the linked worktree build harness resolved `@loaders.gl/compression` to the primary checkout's empty `modules/compression/dist`; this is unrelated to the changed files and did not affect the targeted typecheck, worker build, or test suites.

### CODE REVIEW

File: modules/deck-layers/src/source-layer-utils.ts [hunk 1]
```diff
@@ -406,17 +431,34 @@ function findFirstLeaf(layers: unknown[]): unknown | undefined {
...
   );
 }
 
+/** Finds the first available layer bounds in the same depth-first order as layer selection. */
+function findFirstLayerBounds(
+  layers: unknown[] | undefined
+): [[number, number], [number, number]] | undefined {
+  for (const layer of layers || []) {
+    const typedLayer = layer as {boundingBox?: unknown; layers?: unknown[]};
+    const childBounds = findFirstLayerBounds(typedLayer.layers);
```
[the remark this answers is not in the document]
Comment (ibgreen) [reply modules/deck-layers/src/source-layer-utils.ts]: Addressed in e59d40165c. View-state bounds now use the exact first named leaf selected by automatic layer discovery, so unnamed leaves cannot produce hints for a different rendered layer. Added a regression test covering an unnamed bounded leaf before the selected named leaf.
